test/Microsoft.AspNetCore.Sockets.Client.Tests/ConnectionTests.cs

```diff
@@ -106,6 +106,37 @@ public void ConnectionReturnsUrlUsedToStartTheConnection()
         }
 
         [Fact]
+        public async Task CanStopStartingConnection()
```

Comment:
This is a race so is hard to test. The main idea is that StartAsync is not called on the transport (awaiting the start task would fail since I am using a mock transport). I changed the test a little bit and now I do verify that the `StartAsync` completes before `DisposeAsync` and verify explicitly that `StartAsync` on the transport is not called.